execution_engine/src/storage/trie_store/operations/mod.rs

```diff
@@ -112,6 +115,106 @@ where
     }
 }
 
+/// Same as [`read`], except that a [`TrieMerkleProof`] is generated and returned along with the key
+/// and the value given the root and store.
+#[allow(unused)] // TODO: Use this
+pub fn read_with_proof<K, V, T, S, E>(
+    _correlation_id: CorrelationId,
+    txn: &T,
+    store: &S,
+    root: &Blake2bHash,
+    key: &K,
+) -> Result<ReadResult<TrieMerkleProof<K, V>>, E>
+where
+    K: ToBytes + FromBytes + Eq + std::fmt::Debug,
+    V: ToBytes + FromBytes,
+    T: Readable<Handle = S::Handle>,
+    S: TrieStore<K, V>,
+    S::Error: From<T::Error>,
+    E: From<S::Error> + From<bytesrepr::Error>,
+{
+    let mut proof_steps = VecDeque::new();
```

Comment:
Is there a capacity that we can compute upfront and use `with_capacity()` constructor?

Reply:
Sadly, we can't figure out the capacity before hand.  This vector will contain both pointer blocks and extension nodes, and how many will depend on the data stored in the Trie.